consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinator.java

```diff
@@ -530,17 +530,24 @@ public boolean isDescendantOf(final BlockHeader ancestorBlock, final BlockHeader
         newBlock.getNumber(),
         newBlock.getBlockHash());
 
-    if (ancestorBlock.getBlockHash().equals(newBlock.getHash())) {
+    // start with self, because descending from yourself is valid
+    Optional<BlockHeader> parentOf = Optional.of(newBlock);
+
+    while (parentOf.isPresent()
+        && !parentOf.get().getBlockHash().equals(ancestorBlock.getBlockHash())) {
+      parentOf = protocolContext.getBlockchain().getBlockHeader(parentOf.get().getParentHash());
+    }
+
+    if (parentOf.isPresent()
```

Comment:
I think we can add one extra comparison here and short circuit unrelated forks: check that parentOf.getNumber() is not < ancestor.  That way we don't run all the way back to genesis if we are comparing hashes of blocks on 2 different forks